tool_commands/commands/Step 1 - Check data/schematisation_checker.py

```diff
@@ -0,0 +1,92 @@
+from ThreeDiToolbox.utils.user_messages import pop_up_info
+from ThreeDiToolbox.utils.user_messages import progress_bar
+
+logger = logging.getLogger(__name__)
+
+
+class CustomCommand(CustomCommandBase):
+
+    def __init__(self, iface, ts_datasource, plugin_dir):
+        self.iface = iface
+        self.plugin_dir = plugin_dir
+
+    def show_gui(self):
+        """Show SchemaChecker dialog"""
+        self.modelchecker_widget = SchemaCheckerDialogWidget(
+            self.iface, command=self,
+        )
+        self.modelchecker_widget.exec_()
+
+    def run(self):
+        self.show_gui()
+
+    def run_it(self, threedi_db):
+        """Applies the threedi-modelchecker to `threedi_db`
+
+        The connection to the `threedi_db` and its south_migration_history are first
+        validated. Next, any model errors are written to a text file.
+        """
+        logger.info("Starting threedi-modelchecker")
+        try:
+            model_checker = ThreediModelChecker(threedi_db)
+            model_checker.db.check_connection()
+        except OperationalError as exc:
+            logger.exception("Failed to start a connection with the database.")
+            pop_up_info(
+                "Something went wrong trying to connect to the database, please check"
+                " the connection settings: %s" % exc.args[0]
+            )
+            return
+        except errors.MigrationMissingError:
+            logger.exception("The selected 3Di model does not have the latest "
+                             "migration")
+            pop_up_info(
+                "The selected 3Di model does not have the latest migration, please "
+                "migrate your model to the latest version."
+            )
+            return
+        except errors.MigrationTooHighError:
+            logger.exception(
+                "The selected 3Di model has a higher migration than expected."
+            )
+            pop_up_info(
+                "The 3Di model has a higher migration than expected, do you have "
+                "the latest version of ThreediToolbox?"
+            )
+            return
+        except errors.MigrationNameError:
+            logger.exception(
+                "Unexpected migration name, but migration id is matching. "
+                "We are gonna continue for now and hope for the best."
+            )
+
+        output_filename = 'Model_errors.txt'
+        output_file_path = os.path.join(self.plugin_dir, output_filename)
+
+        session = model_checker.db.get_session()
+
+        total_checks = len(model_checker.config.checks)
+        with progress_bar(self.iface, max_value=total_checks) as pb, \
+                open(output_file_path, "w") as output_file:
+            for i, check in enumerate(model_checker.checks()):
+                model_errors = check.get_invalid(session)
+                for error_row in model_errors:
+                    formatted_error = format_check_results(check, error_row)
```

Comment:
Even nakijken of hier de verantwoordelijkheden netjes zijn gescheiden.
In deze `schematisation_checker.py` voer je de resultaten van de eigenlijke check (die uit een andere library komt) aan een `format_check_results()` die weer uit een andere file komt.

Hoe gaat dat als je een veldje in de check toevoegt of weghaalt? Dat soort vragen. Kan de library al niet een geformatteerde "msg" aan de row toevoegen?

Brainstorm: in hoeverre is een csv output i.p.v. een .txt nog handig? Ik weet niet welke fouten worden gerapporteerd, maar misschien helpt een csv met sorteren enzo.